apps/api_web/lib/api_web/plugs/rate_limiter.ex

```diff
@@ -11,15 +11,32 @@ defmodule ApiWeb.Plugs.RateLimiter do
 
   def call(%{assigns: assigns, request_path: request_path} = conn, _) do
     case ApiWeb.RateLimiter.log_request(assigns.api_user, request_path) do
-      :ok ->
+      %{status: :ok, limit_data: %{limit: limit, remaining: remaining, reset_ms: reset_ms}} ->
```

Comment:
You don't really need a map for this: `{:ok | :rate_limited, limit_data} | :ok`. `limit_data` itself could be a tuple too, `{limit, remaining, reset_ms}`